plugin/src/test/scala/com/typesafe/genjavadoc/util/GenJavaDocCompiler.scala

```diff
@@ -0,0 +1,38 @@
+package com.typesafe.genjavadoc
+package util
+
+import scala.io.Source
+import scala.reflect.io.AbstractFile
+import scala.tools.nsc.reporters.ConsoleReporter
+import scala.tools.nsc.{Global, Settings}
+
+/** An instance of the scala compiler with the genjavadoc plugin enabled
+  * @param params additional parameters to pass to the compiler
+  */
+class GenJavaDocCompiler(params: String*) {
```

Comment:
This is the central new feature: all tests have been updated to use programmatic instantiation of the Scala compiler (instead of relying on sbt)